src/lib/fetch.ts

```diff
@@ -91,11 +102,42 @@ export async function azleFetch(input: any, init?: any): Promise<any> {
             text: async () => {
                 return contents.toString();
             }
-        };
+        } as any;
     }
 
-    if (url.protocol === 'http:') {
-        // TODO http call to the management canister
+    if (url.protocol === 'https:' || url.protocol === 'http:') {
+        const body = await prepareRequestBody(init);
+        // TODO also do headers and method and everything like on the client?
+
+        const response = await azleFetch(`icp://aaaaa-aa/http_request`, {
+            body: serialize({
+                args: [
+                    {
+                        url: input,
+                        max_response_bytes: getHttpMaxResponseBytes(),
+                        method: getHttpMethod(init),
+                        headers: init?.headers ?? [],
+                        body,
+                        transform: getHttpTransform()
+                    }
+                ],
+                cycles:
+                    globalThis._azleOutgoingHttpOptionsCycles ?? 3_000_000_000n // TODO this seems to be a conservative max size
```

Comment:
Do we have any calculations that help us arrive at 3_000_000_000n or is that just a best guess? or found through trial and error?

Reply:
I have a cycles calculation, it works and is accurate according to the documentation, but ~20x more expensive than it should be according to the errors from the IC itself ($.02 per call instead of 20x less). It's just a default, I have asked in the forum to see if we can figure out why this is the case: https://forum.dfinity.org/t/a-new-price-function-for-https-outcalls/20838/21

Reply:
Apparently I have it correct, the price I am calculating is the 13 node subnet price...we'll just leave it hardcoded like this for now then, I have added the ability to set the subnet size with the global http call settings (or I will add that)

Reply:
I added the ability to change subnet size